File: lib/domain/models/tour.dart
import 'coordinates.dart';

enum TourStatus { planned, completed }

class Tour {
  final int id;
  final DateTime plannedDate;
  final int startTimeMinutes;
  final TourStatus status;
  final int totalDistanceMeters;
  final int totalDriveSeconds;
  final int totalTravelFeeCents;
  final String? notes;
  final DateTime? completedAt;
  final DateTime createdAt;

  /// Polyline ORS de la tournée (base → stops dans l'ordre → base).
  /// Nullable : si non disponible (réseau au moment de la création, ou
  /// tournée pré-feature), le rendu carte tombe en lignes droites.
  final List<Coordinates>? routeGeometry;

  const Tour({
    required this.id,
    required this.plannedDate,
    required this.startTimeMinutes,
    required this.status,
    required this.totalDistanceMeters,
    required this.totalDriveSeconds,
    required this.totalTravelFeeCents,
    required this.createdAt,
    this.notes,
    this.completedAt,
    this.routeGeometry,
  });
}

File: src/domain/models/tour.ts
import { z } from 'zod';

export const TourStatus = z.enum(['planned', 'completed']);
export type TourStatus = z.infer<typeof TourStatus>;

export const Tour = z.object({
  id: z.string(),
  scheduledDate: z.string(),
  departureTime: z.string(),
  baseLat: z.number(),
  baseLng: z.number(),
  status: TourStatus,
  totalDistanceKm: z.number().nullable(),
  totalDriveSeconds: z.number().int().nullable(),
  totalMinutes: z.number().int().nullable(),
  totalRevenueCents: z.number().int().nullable(),
  totalAnimalsCount: z.number().int().nullable(),
  routeGeometry: z.string().nullable(),
  notes: z.string().nullable(),
  completedAt: z.string().nullable(),
  createdAt: z.string(),
  updatedAt: z.string(),
});

export type Tour = z.infer<typeof Tour>;
